Spaced Comment Exceptions Not Working

I have the following config

```
"spaced-comment": [2, "always", { exceptions: ["*", "-"] }],
```

I'm trying the following code:

```
/**************
 * Private API
 **************/
```

It's giving me the following error:

```
models/abcModel.es6
  12:1  error  Expected exception block, space or tab after /* in comment  spaced-comment
```

I would assume that this would  **not warn**, because the docs give a similar example. Maybe it's an issue with `*` or something....

--- a/docs/rules/spaced-comment.md
+++ b/docs/rules/spaced-comment.md
@@ -4,128 +4,155 @@ Some style guides require or disallow a whitespace immediately after the initial
 Whitespace after the `//` or `/*` makes it easier to read text in comments.
 On the other hand, commenting out code is easier without having to put a whitespace right after the `//` or `/*`.
 
-
 ## Rule Details
 
-This rule will enforce consistency of spacing after the start of a comment `//` or `/*`.
+This rule will enforce consistency of spacing after the start of a comment `//` or `/*`. It also provides several
+exceptions for various documentation styles.
 
-This rule takes two arguments. If the first is `"always"` then the `//` or `/*` must be followed by at least once whitespace.
-If `"never"` then there should be no whitespace following.
-The default is `"always"`.
+### Options
 
-The second argument is an object with two keys, `"exceptions"` and `"markers"`.
+The rule takes two options. The first is a string which be either "always" or "never". If you pass `"always"` then the `//` or `/*` must be followed by at least once whitespace. If `"never"` then there should be no whitespace following. The default is `"always"`.
+
+Here is an example of how to configure the rule with this option:
+
+```js
+"spaced-comment": [2, "always"]
+```
+
+#### Exceptions
+
+This rule can also take a 2nd option, an object with either of the following keys: `"exceptions"` and `"markers"`.
+
 The `"exceptions"` value is an array of string patterns which are considered exceptions to the rule.
-It is important to note that the exceptions are ignored if the first argument is `"never"`.
+Please note that exceptions are ignored if the first argument is `"never"`.
 
-Exceptions cannot be mixed. From the collection of exceptions provided only one of them can be used inside the comment. Mixing of more than one is not valid.
+```js
+"spaced-comment": [2, "always", { "exceptions": ["-", "+"] }]
+```
 
 The `"markers"` value is an array of string patterns which are considered markers for docblock-style comments,
 such as an additional `/`, used to denote documentation read by doxygen, vsdoc, etc. which must have additional characters.
 The `"markers"` array will apply regardless of the value of the first argument, e.g. `"always"` or `"never"`.
 
-The following patterns are considered warnings:
+```js
+"spaced-comment": [2, "always", { "markers": ["/"] }]
+```
+
+The difference between a marker and an exception is that a marker only appears at the beginning of the comment whereas
+exceptions can occur anywhere in the comment string.
+
+#### Examples
+
+The following patterns **are** considered warnings:
 
 ```js
-// When ["never"]
+// When [2, "never"]
 // This is a comment with a whitespace at the beginning
 ```
 
 ```js
-// When ["never"]
+// When [2, "never"]
 /* This is a comment with a whitespace at the beginning */
 ```
 
+
 ```js
-// When ["never"]
+// When [2, "never"]
 /* \nThis is a comment with a whitespace at the beginning */
 ```
 
 ```js
-//When ["always"]
+// When [2, "always"]
 //This is a comment with no whitespace at the beginning
 var foo = 5;
 ```
 
 ```js
-//When ["always"]
+// When [2, "always"]
 /*This is a comment with no whitespace at the beginning */
 var foo = 5;
 ```
 
 ```js
-// When ["always",{"exceptions":["-","+"]}]
+// When [2, "always", { "exceptions": ["-", "+"] }]
 //------++++++++
 // Comment block
 //------++++++++
 ```
 
 ```js
-// When ["always",{"markers":["/"]}]
+// When [2, "always", { "markers": ["/"] }]
 ///This is a comment with a marker but without whitespace
 ```
 
 ```js
-// When ["always",{"exceptions":["-","+"]}]
+// When [2, "always", { "exceptions": ["-", "+"] }]
 /*------++++++++*/
 /* Comment block */
 /*------++++++++*/
 ```
 
-The following patterns are not warnings:
+The following patterns **are not** warnings:
 
 ```js
-// When ["always"]
+// When [2, "always"]
 // This is a comment with a whitespace at the beginning
 var foo = 5;
 ```
 
 ```js
-// When ["always"]
+// When [2, "always"]
 /* This is a comment with a whitespace at the beginning */
 var foo = 5;
 ```
 
 ```js
-// When ["always"]
+// When [2, "always"]
 /*\n * This is a comment with a whitespace at the beginning */
 var foo = 5;
 ```
 
-
 ```js
-//When ["never"]
+// When [2, "never"]
 /*This is a comment with no whitespace at the beginning */
 var foo = 5;
 ```
 
 ```js
-// When ["always",{"exceptions":["-"]}]
+// When [2, "always", { "exceptions": ["-"] }]
 //--------------
 // Comment block
 //--------------
 ```
 
 ```js
-// When ["always",{"exceptions":["-+"]}]
+// When [2, "always", { "exceptions": ["-+"] }]
 //-+-+-+-+-+-+-+
 // Comment block
 //-+-+-+-+-+-+-+
 ```
 
 ```js
-// When ["always",{"exceptions":["-+"]}]
+// When [2, "always", { "exceptions": ["-+"] }]
 /*-+-+-+-+-+-+-+*/
 // Comment block
 /*-+-+-+-+-+-+-+*/
 ```
 
 ```js
-// When ["always",{"markers":["/"]}]
+// When [2, "always", { "exceptions": ["*"] }]
+/****************
+ * Comment block
+ ****************/
+```
+
+```js
+// When [2, "always", { "markers": ["/"] }]
 /// This is a comment with a marker
 ```
 
 ```js
-// When ["never",{"markers":["!<"]}]
+// When [2, "never", { "markers": ["!<"] }]
 //!<This is a comment with a marker
 /*!<this is a block comment with a marker
 subsequent lines are ignored
@@ -133,14 +160,19 @@ subsequent lines are ignored
 ```
 
 ```js
-// When ["always"]
+// When [2, "always", { "markers": ["global"] }]
+/*global ABC*/
+```
+
+```js
+// When [2, "always"]
 /**
 * I am jsdoc
 */
 ```
 
 ```js
-// When ["never"]
+// When [2, "never"]
 /**
 * I am jsdoc
 */

--- a/lib/rules/spaced-comment.js
+++ b/lib/rules/spaced-comment.js
@@ -111,9 +111,8 @@ function createAlwaysStylePattern(markers, exceptions) {
             pattern += exceptions.map(escapeAndRepeat).join("|");
             pattern += ")";
         }
-        pattern += "$)"; // the sequence continues until the end.
+        pattern += "(?:$|[\n\r]))"; // the sequence continues until the end.
     }
-
     return new RegExp(pattern);
 }
 

--- a/tests/lib/rules/spaced-comment.js
+++ b/tests/lib/rules/spaced-comment.js
@@ -28,6 +28,8 @@ ruleTester.run("spaced-comment", rule, {
             code: "//A valid comment NOT starting with space\nvar a = 2;",
             options: ["never"]
         },
+
+        // exceptions - line comments
         {
             code: "//-----------------------\n// A comment\n//-----------------------",
             options: ["always", {
@@ -52,6 +54,34 @@ ruleTester.run("spaced-comment", rule, {
                 exceptions: ["-", "=", "*", "#", "!@#"]
             }]
         },
+
+        // exceptions - block comments
+        {
+            code: "var a = 1; /*######*/",
+            options: ["always", {
+                exceptions: ["-", "=", "*", "#", "!@#"]
+            }]
+        },
+        {
+            code: "/*****************\n * A comment\n *****************/",
+            options: ["always", {
+                exceptions: ["*"]
+            }]
+        },
+        {
+            code: "/*++++++++++++++\n * A comment\n +++++++++++++++++*/",
+            options: ["always", {
+                exceptions: ["+"]
+            }]
+        },
+        {
+            code: "/*++++++++++++++\n + A comment\n * B comment\n - C comment\n----------------*/",
+            options: ["always", {
+                exceptions: ["+", "-"]
+            }]
+        },
+
+        // markers - line comments
         {
             code: "//!< docblock style comment",
             options: ["always", {
@@ -71,6 +101,40 @@ ruleTester.run("spaced-comment", rule, {
                 markers: ["/", "!<"]
             }]
         },
+
+        // markers - block comments
+        {
+            code: "var a = 1; /*# This is an example of a marker in a block comment\nsubsequent lines do not count*/",
+            options: ["always", {
+                markers: ["#"]
+            }]
+        },
+        {
+            code: "/*!\n *comment\n */",
+            options: ["always", { markers: ["!"] }]
+        },
+        {
+            code: "/**\n *jsdoc\n */",
+            options: ["always", { markers: ["*"] }]
+        },
+        {
+            code: "/*global ABC*/",
+            options: ["always", { markers: ["global"] }]
+        },
+        {
+            code: "/*eslint-env node*/",
+            options: ["always", { markers: ["eslint-env"] }]
+        },
+        {
+            code: "/*eslint eqeqeq:0, curly: 2*/",
+            options: ["always", { markers: ["eslint"] }]
+        },
+        {
+            code: "/*eslint-disable no-alert, no-console */\nalert()\nconsole.log()\n/*eslint-enable no-alert */",
+            options: ["always", { markers: ["eslint-enable", "eslint-disable"] }]
+        },
+
+        // misc. variations
         {
             code: validShebangProgram,
             options: ["always"]
@@ -91,18 +155,8 @@ ruleTester.run("spaced-comment", rule, {
             code: "var a = 1; /* A valid comment starting with space */",
             options: ["always"]
         },
-        {
-            code: "var a = 1; /*######*/",
-            options: ["always", {
-                exceptions: ["-", "=", "*", "#", "!@#"]
-            }]
-        },
-        {
-            code: "var a = 1; /*# This is an example of a marker in a block comment\nsubsequent lines do not count*/",
-            options: ["always", {
-                markers: ["#"]
-            }]
-        },
+
+        // block comments
         {
             code: "var a = 1; /*A valid comment NOT starting with space */",
             options: ["never"]
@@ -140,16 +194,8 @@ ruleTester.run("spaced-comment", rule, {
             options: ["always"]
         },
         {
-            code: "/*!\n *comment\n */",
-            options: ["always", { markers: ["!"] }]
-        },
-        {
             code: "/**\n *jsdoc\n */",
             options: ["always"]
-        },
-        {
-            code: "/**\n *jsdoc\n */",
-            options: ["always", { markers: ["*"] }]
         },
         {
             code: "/**\r\n *jsdoc\r\n */",
@@ -164,6 +210,7 @@ ruleTester.run("spaced-comment", rule, {
             options: ["always"]
         },
 
+        // markers & exceptions
         {
             code: "///--------\r\n/// test\r\n///--------",
             options: ["always", {markers: ["/"], exceptions: ["-"]}]

